license-details/README.md

```diff
@@ -0,0 +1,138 @@
+# Introduction
+
+The P4 compiler `p4c` is a leading project of the P4 Consortium, and
+used by several companies as the front end of P4 compilers for a
+variety of hardware targets.  Most of these companies develop
+significant proprietary back end code, and the resulting binary
+executables are released, with no copyleft obligation for those
+companies to release their back end source code.
+
+This is allowed, because the `p4c` front source files are licensed
+under the Apache 2.0 license.  This license was chosen for multiple
+reasons, but in part to enable companies to do this.  It is the P4
+Consortium's intent to continue enabling this.
+
+
+# Linux Foundation review of P4.org software licenses in published code
+
+In early 2024 P4.org became part of The Linux Foundation.  In November
+2024 the Linux Foundation appointed someone to do a quick review of
+software licenses in the public repositories in the
+https://github.com/p4lang organization.
+
+They recommended a fairly light-weight mechanism to make the software
+license of all source files explicit and easy to track with automated
+tools: adding an SPDX-License-Identifier comment line [1].
+
+[1] https://spdx.org/licenses/
+
+These look like the following examples, where the syntax of everything
+_except_ the comment indicator is required by SPDX, i.e. there is a
+precise specified syntax for ethe "SPDX-License-Identifier:"
+characters to be present, and exactly what sequence of characters may
+follow it on the line.
+
+```c++
+// SPDX-License-Identifier: Apache-2.0
+```
+
+```python
+# SPDX-License-Identifier: GPL-2.0-only
+```
+
+
+# Why is the top level LICENSE file insufficient?
+
+One short answer is that not all files in this repository can legally
```

Comment:
This model sounds questionable to me. This should be a big exception. Similar to the inbound-outbound model LLVM has: https://llvm.org/docs/DeveloperPolicy.html#copyright-license-and-patents

Reply:
This topic is on the agenda for the next P4 TST meeting on 2025-Feb-07, and seems worth discussing there.  I am glad if this early draft gets better plans made, before we spend too much time on what it currently describes.